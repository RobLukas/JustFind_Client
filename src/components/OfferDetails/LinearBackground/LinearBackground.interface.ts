import TechnologyType from 'api/technologyCollection';

interface LinearBackgroundProps {
  technology: TechnologyType;
}

export default LinearBackgroundProps;
