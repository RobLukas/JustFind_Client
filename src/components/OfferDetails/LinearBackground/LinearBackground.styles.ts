import styled from 'styled-components';

interface IProps {
  backgroundColor: string;
}

const LinearBackground = styled.div<IProps>(({ backgroundColor }) => ({
  background: `linear-gradient(${backgroundColor})`,
  width: 5,
  borderRadius: '6px 0 0 6px',
}));

export default {
  LinearBackground,
};
