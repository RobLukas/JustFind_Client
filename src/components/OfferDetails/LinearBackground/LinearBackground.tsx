import React, { FC } from 'react';
import linearBackgroundTechCollection from '@styles/linearBackgroundTechEnum';
import LinearBackgroundProps from './LinearBackground.interface';
import S from './LinearBackground.styles';

const LinearBackground: FC<LinearBackgroundProps> = ({
  technology,
}: LinearBackgroundProps) => {
  return (
    <S.LinearBackground
      backgroundColor={linearBackgroundTechCollection[technology]}
    />
  );
};

export default LinearBackground;
