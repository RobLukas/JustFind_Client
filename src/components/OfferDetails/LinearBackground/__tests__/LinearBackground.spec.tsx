import { cleanup, render } from '@testing-library/react';
import React from 'react';
import LinearBackground from '../LinearBackground';
import LinearBackgroundProps from '../LinearBackground.interface';

describe('<LinearBackground />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const { technology }: LinearBackgroundProps = { technology: 'JS' };

    const { container } = render(<LinearBackground technology={technology} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
