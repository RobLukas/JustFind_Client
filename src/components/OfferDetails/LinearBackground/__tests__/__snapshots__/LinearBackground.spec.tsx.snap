// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinearBackground /> render correctly 1`] = `
<div
  class="sc-bdnylx jpNXAy"
/>
`;
